modules/gdscript/gd_functions.cpp

```diff
@@ -816,6 +818,81 @@ void GDFunctions::call(Function p_func,const Variant **p_args,int p_arg_count,Va
 			}
 
 		} break;
+		case GEN_XRANGE: {
+
+			switch(p_arg_count) {
+				case 0: {
+					r_error.error=Variant::CallError::CALL_ERROR_TOO_FEW_ARGUMENTS;
+					r_error.argument=1;
+				} break;
+				case 1: {
+
+					VALIDATE_ARG_NUM(0);
+
+					int count=*p_args[0];
+
+					RangeIterator *itr = memnew(RangeIterator);
```

Comment:
It leaks, yes. I made it a Reference because I couldn't find any convenient location to call `memdelete`. To fix it, replace the allocations with Refs:
```GDScript
Ref<RangeIterator> itr = Ref<RangeIterator>( memnew(RangeIterator) );
```